packages/core/src/anchor/ethereum/ethereum-anchor-service.ts

```diff
@@ -235,10 +253,17 @@ export default class EthereumAnchorService extends AnchorService {
             throw new Error('Invalid chain ID according to CAIP-2')
         }
 
+        if (!this._supportedChains.includes(chain)) {
```

Comment:
This line seems redundant? this._supportedChains is the CAS chains and chain is one of the CAS chains from the loop.

Reply:
`chain` can also come from an anchorProof that needs validating, which was discovered from an anchorCommit and could theoretically be any chain if the user anchored manually

Reply:
Maybe leave the check here, and remove it from `init` then?

Reply:
Not sure I follow.  The checks in `init` are different.  Which part are you suggesting is unnecessary?

Reply:
OHSHI. Sorry. They really are different. Please, do not mind.